packages/shared/src/variants/baduk.ts

```diff
@@ -43,18 +56,23 @@ export class Baduk extends AbstractGame<BadukConfig, BadukState> {
       throw new Error("Baduk does not support sizes greater than 52");
     }
 
-    this.board = new Grid<Color>(this.config.width, this.config.height).fill(
-      Color.EMPTY,
-    );
+    if (isGridBadukConfig(this.config)) {
+      this.board = new Grid<Color>(this.config.width, this.config.height).fill(
+        Color.EMPTY,
+      );
+    } else {
+      this.boardAsIntersections = createBoard(this.config.board!, Intersection);
+      this.board = new GraphWrapper(createGraph(this.boardAsIntersections));
+    }
   }
 
   override exportState(): BadukState {
     return {
-      board: this.board.to2DArray(),
```

Comment:
I'd really like to keep the exports "JSON-compatible".  What I mean by that is that information should not be lost by converting to JSON:

```
let grid: Grid<number> = ...;
let arr: number[][] = ...;

grid = JSON.parse(JSON.stringify(grid));  // not actually a Grid
arr = JSON.parse(JSON.stringify(arr)); // a 2D Array, just like the original
```

**Motivation**:  We need to keep in mind that game state needs to travel over the wire at some point (#48, #174), and we need a way to losslessly encode the state.

In my mind, the way to do that is to return simple types from `exportState()` with helper methods like `Grid.prototype.to2DArray()`, then use the JSON utils to encode and decode.  (If you need a `Graph`-compatible way to do this, feel free to alias that method to `Grid.prototype.serialize()`).

And let me know if you'd propose another way to pass state over the wire that doesn't simplification inside exportState!